### ISSUE TITLE
[Support]: MQTT zone object counts not reflecting detection events

### ISSUE THREAD

**Issue by jmmyerz** [blakeblackshear/frigate#3819]
### Describe the problem you are having

![image](https://user-images.githubusercontent.com/12993229/189547634-8316b3a9-99c4-4db7-91be-ad6240168795.png)

Zone object counts in MQTT do not always accurately reflect the detected objects in a zone (as seen in event data or debug views). The image above should (I believe) count 1 car in the zone based on confidence, but It's counting 0. The count will intermittently change to 1 if the vehicle on the left's confidence score increases, or if the white car's bounding box changes. Both car objects are occasionally counted, but it is not always accurate even when they are detected as having entered or are currently in a zone.

### Version

0.11.0-c461c9e

### Frigate config file

```yaml
{
  "birdseye": {
    "enabled": true,
    "height": 720,
    "mode": "objects",
    "quality": 8,
    "width": 1280
  },
  "cameras": {
    "garage_east": {
      "best_image_timeout": 60,
      "birdseye": {
        "enabled": true,
        "mode": "objects"
      },
      "detect": {
        "enabled": true,
        "fps": 5,
        "height": 720,
        "max_disappeared": 25,
        "stationary": {
          "interval": 0,
          "max_frames": {
            "default": null,
            "objects": {}
          },
          "threshold": 50
        },
        "width": 1280
      },
      "ffmpeg": {
        "global_args": [
          "-hide_banner",
          "-loglevel",
          "warning"
        ],
        "hwaccel_args": [
          "-hwaccel",
          "vaapi",
          "-hwaccel_device",
          "/dev/dri/renderD128",
          "-hwaccel_output_format",
          "yuv420p"
        ],
        "input_args": [
          "-avoid_negative_ts",
          "make_zero",
          "-fflags",
          "+genpts+discardcorrupt",
          "-rtsp_transport",
          "tcp",
          "-timeout",
          "5000000",
          "-use_wallclock_as_timestamps",
          "1"
        ],
        "inputs": [
          {
            "global_args": [],
            "hwaccel_args": [],
            "input_args": [],
            "path": "rtsp://frigate:[redacted]@10.13.2.221:554/Streaming/Channels/101",
            "roles": [
              "record"
            ]
          },
          {
            "global_args": [],
            "hwaccel_args": [],
            "input_args": [],
            "path": "rtsp://frigate:[redacted]@10.13.2.221:554/Streaming/Channels/102",
            "roles": [
              "rtmp"
            ]
          },
          {
            "global_args": [],
            "hwaccel_args": [],
            "input_args": [],
            "path": "rtsp://frigate:[redacted]@10.13.2.221:554/Streaming/Channels/103",
            "roles": [
              "detect"
            ]
          }
        ],
        "output_args": {
          "detect": [
            "-f",
            "rawvideo",
            "-pix_fmt",
            "yuv420p"
          ],
          "record": "-f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c:v copy -c:a aac",
          "rtmp": "-c:v copy -c:a aac -f flv"
        }
      },
      "ffmpeg_cmds": [
        {
          "cmd": "ffmpeg -hide_banner -loglevel warning -hwaccel vaapi -hwaccel_device /dev/dri/renderD128 -hwaccel_output_format yuv420p -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -timeout 5000000 -use_wallclock_as_timestamps 1 -i rtsp://frigate:[redacted]@10.13.2.221:554/Streaming/Channels/101 -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c:v copy -c:a aac /tmp/cache/garage_east-%Y%m%d%H%M%S.mp4",
          "roles": [
            "record"
          ]
        },
        {
          "cmd": "ffmpeg -hide_banner -loglevel warning -hwaccel vaapi -hwaccel_device /dev/dri/renderD128 -hwaccel_output_format yuv420p -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -timeout 5000000 -use_wallclock_as_timestamps 1 -i rtsp://frigate:[redacted]@10.13.2.221:554/Streaming/Channels/102 -c:v copy -c:a aac -f flv rtmp://127.0.0.1/live/garage_east",
          "roles": [
            "rtmp"
          ]
        },
        {
          "cmd": "ffmpeg -hide_banner -loglevel warning -hwaccel vaapi -hwaccel_device /dev/dri/renderD128 -hwaccel_output_format yuv420p -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -timeout 5000000 -use_wallclock_as_timestamps 1 -i rtsp://frigate:[redacted]@10.13.2.221:554/Streaming/Channels/103 -r 5 -s 1280x720 -f rawvideo -pix_fmt yuv420p pipe:",
          "roles": [
            "detect"
          ]
        }
      ],
      "live": {
        "height": 720,
        "quality": 1
      },
      "motion": {
        "contour_area": 30,
        "delta_alpha": 0.2,
        "frame_alpha": 0.2,
        "frame_height": 50,
        "improve_contrast": false,
        "mask": [
          "1280,0,1280,343,0,387,0,0"
        ],
        "mqtt_off_delay": 30,
        "threshold": 25
      },
      "mqtt": {
        "bounding_box": true,
        "crop": true,
        "enabled": true,
        "height": 270,
        "quality": 70,
        "required_zones": [],
        "timestamp": true
      },
      "name": "garage_east",
      "objects": {
        "filters": {
          "bicyle": {
            "mask": null,
            "max_area": 24000000,
            "max_ratio": 24000000,
            "min_area": 0,
            "min_ratio": 0,
            "min_score": 0.5,
            "threshold": 0.7
          },
          "car": {
            "mask": null,
            "max_area": 24000000,
            "max_ratio": 24000000,
            "min_area": 0,
            "min_ratio": 0,
            "min_score": 0.5,
            "threshold": 0.7
          },
          "cat": {
            "mask": null,
            "max_area": 24000000,
            "max_ratio": 24000000,
            "min_area": 0,
            "min_ratio": 0,
            "min_score": 0.5,
            "threshold": 0.7
          },
          "dog": {
            "mask": null,
            "max_area": 24000000,
            "max_ratio": 24000000,
            "min_area": 0,
            "min_ratio": 0,
            "min_score": 0.5,
            "threshold": 0.7
          },
          "motorcycle": {
            "mask": null,
            "max_area": 24000000,
            "max_ratio": 24000000,
            "min_area": 0,
            "min_ratio": 0,
            "min_score": 0.5,
            "threshold": 0.7
          },
          "person": {
            "mask": null,
            "max_area": 24000000,
            "max_ratio": 24000000,
            "min_area": 0,
            "min_ratio": 0,
            "min_score": 0.5,
            "threshold": 0.7
          }
        },
        "mask": "",
        "track": [
          "person",
          "cat",
          "dog",
          "bicyle",
          "motorcycle",
          "car"
        ]
      },
      "record": {
        "enabled": true,
        "events": {
          "objects": null,
          "post_capture": 5,
          "pre_capture": 5,
          "required_zones": [
            "driveway"
          ],
          "retain": {
            "default": 14,
            "mode": "active_objects",
            "objects": {}
          }
        },
        "expire_interval": 60,
        "retain": {
          "days": 2,
          "mode": "all"
        },
        "retain_days": null
      },
      "rtmp": {
        "enabled": true
      },
      "snapshots": {
        "bounding_box": true,
        "clean_copy": true,
        "crop": false,
        "enabled": true,
        "height": null,
        "quality": 70,
        "required_zones": [],
        "retain": {
          "default": 30,
          "mode": "motion",
          "objects": {}
        },
        "timestamp": false
      },
      "timestamp_style": {
        "color": {
          "blue": 255,
          "green": 255,
          "red": 255
        },
        "effect": null,
        "format": "%m/%d/%Y %H:%M:%S",
        "position": "tl",
        "thickness": 2
      },
      "ui": {
        "dashboard": true,
        "order": 0
      },
      "zones": {
        "driveway": {
          "coordinates": "0,720,1280,720,1280,459,1105,359,0,406",
          "filters": {},
          "objects": []
        }
      }
    }
  },
  "database": {
    "path": "/media/frigate/frigate.db"
  },
  "detect": {
    "enabled": true,
    "fps": 5,
    "height": 720,
    "max_disappeared": null,
    "stationary": {
      "interval": 0,
      "max_frames": {
        "default": null,
        "objects": {}
      },
      "threshold": null
    },
    "width": 1280
  },
  "detectors": {
    "coral": {
      "device": "pci:0",
      "num_threads": 3,
      "type": "edgetpu"
    }
  },
  "environment_vars": {},
  "ffmpeg": {
    "global_args": [
      "-hide_banner",
      "-loglevel",
      "warning"
    ],
    "hwaccel_args": [
      "-hwaccel",
      "vaapi",
      "-hwaccel_device",
      "/dev/dri/renderD128",
      "-hwaccel_output_format",
      "yuv420p"
    ],
    "input_args": [
      "-avoid_negative_ts",
      "make_zero",
      "-fflags",
      "+genpts+discardcorrupt",
      "-rtsp_transport",
      "tcp",
      "-timeout",
      "5000000",
      "-use_wallclock_as_timestamps",
      "1"
    ],
    "output_args": {
      "detect": [
        "-f",
        "rawvideo",
        "-pix_fmt",
        "yuv420p"
      ],
      "record": "-f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c:v copy -c:a aac",
      "rtmp": "-c:v copy -c:a aac -f flv"
    }
  },
  "live": {
    "height": 720,
    "quality": 1
  },
  "logger": {
    "default": "info",
    "logs": {
      "frigate.app": "debug"
    }
  },
  "model": {
    "height": 320,
    "labelmap": {},
    "labelmap_path": null,
    "path": null,
    "width": 320
  },
  "motion": {
    "contour_area": 30,
    "delta_alpha": 0.2,
    "frame_alpha": 0.2,
    "frame_height": 50,
    "improve_contrast": false,
    "mask": [
      "0,355,1280,319,1280,0,0,0"
    ],
    "mqtt_off_delay": 30,
    "threshold": 25
  },
  "mqtt": {
[redacted]
    "port": 1883,
    "stats_interval": 60,
    "tls_ca_certs": null,
    "tls_client_cert": null,
    "tls_client_key": null,
    "tls_insecure": null,
    "topic_prefix": "frigate",
    "user": "frigate"
  },
  "objects": {
    "filters": null,
    "mask": "",
    "track": [
      "person",
      "cat",
      "dog",
      "bicyle",
      "motorcycle",
      "car"
    ]
  },
  "plus": {
    "enabled": true
  },
  "record": {
    "enabled": true,
    "events": {
      "objects": null,
      "post_capture": 5,
      "pre_capture": 5,
      "required_zones": [],
      "retain": {
        "default": 14,
        "mode": "active_objects",
        "objects": {}
      }
    },
    "expire_interval": 60,
    "retain": {
      "days": 2,
      "mode": "all"
    },
    "retain_days": null
  },
  "rtmp": {
    "enabled": true
  },
  "snapshots": {
    "bounding_box": true,
    "clean_copy": true,
    "crop": false,
    "enabled": true,
    "height": null,
    "quality": 70,
    "required_zones": [],
    "retain": {
      "default": 30,
      "mode": "motion",
      "objects": {}
    },
    "timestamp": false
  },
  "timestamp_style": {
    "color": {
      "blue": 255,
      "green": 255,
      "red": 255
    },
    "effect": null,
    "format": "%m/%d/%Y %H:%M:%S",
    "position": "tl",
    "thickness": 2
  },
  "ui": {
    "use_experimental": false
  }
}
```

### Relevant log output

```shell
N/A
```

### FFprobe output from your camera

```shell
Input #0, rtsp, from 'rtsp://frigate:Streampw@10.13.2.221:554/Streaming/Channels/103':
  Metadata:
    title           : Media Presentation
  Duration: N/A, start: 0.742000, bitrate: N/A
    Stream #0:0: Video: h264 (Main), yuvj420p(pc, bt709, progressive), 1280x720 [SAR 1:1 DAR 16:9], 8 fps, 30 tbr, 90k tbn, 16 tbc
```

### Operating system

Debian

### Install method

Docker Compose

### Coral version

M.2

### Network connection

Wired

### Camera make and model

hikvision DS-2CD2563G0-IS

### Any other information that may be helpful

Event data when the white car is counted inside the driveway zone:
```json
{
  "before": {
    "id": "1662924869.841532-945a8u",
    "camera": "garage_east",
    "frame_time": 1662925743.928876,
    "snapshot_time": 1662924873.926731,
    "label": "car",
    "top_score": 0.75390625,
    "false_positive": false,
    "start_time": 1662924869.841532,
    "end_time": null,
    "score": 0.75390625,
    "box": [
      307,
      295,
      882,
      717
    ],
    "area": 242650,
    "ratio": 1.3625592417061612,
    "region": [
      203,
      0,
      999,
      796
    ],
    "stationary": true,
    "motionless_count": 4351,
    "position_changes": 1,
    "current_zones": [
      "driveway"
    ],
    "entered_zones": [
      "driveway"
    ],
    "has_clip": true,
    "has_snapshot": true
  },
  "after": {
    "id": "1662924869.841532-945a8u",
    "camera": "garage_east",
    "frame_time": 1662925804.287795,
    "snapshot_time": 1662924873.926731,
    "label": "car",
    "top_score": 0.75390625,
    "false_positive": false,
    "start_time": 1662924869.841532,
    "end_time": null,
    "score": 0.75390625,
    "box": [
      307,
      295,
      882,
      717
    ],
    "area": 242650,
    "ratio": 1.3625592417061612,
    "region": [
      203,
      0,
      999,
      796
    ],
    "stationary": true,
    "motionless_count": 4653,
    "position_changes": 1,
    "current_zones": [
      "driveway"
    ],
    "entered_zones": [
      "driveway"
    ],
    "has_clip": true,
    "has_snapshot": true
  },
  "type": "update"
}
```

Event data when the white car is _not_ counted in the driveway zone:
```json
{
  "before": {
    "id": "1662925983.809688-2kf62b",
    "camera": "garage_east",
    "frame_time": 1662925988.655901,
    "snapshot_time": 1662925988.655901,
    "label": "car",
    "top_score": 0.796875,
    "false_positive": false,
    "start_time": 1662925983.809688,
    "end_time": null,
    "score": 0.71484375,
    "box": [
      200,
      275,
      880,
      731
    ],
    "area": 310080,
    "ratio": 1.4912280701754386,
    "region": [
      54,
      0,
      926,
      872
    ],
    "stationary": false,
    "motionless_count": 10,
    "position_changes": 1,
    "current_zones": [],
    "entered_zones": [
      "driveway"
    ],
    "has_clip": true,
    "has_snapshot": true
  },
  "after": {
    "id": "1662925983.809688-2kf62b",
    "camera": "garage_east",
    "frame_time": 1662925996.920892,
    "snapshot_time": 1662925988.655901,
    "label": "car",
    "top_score": 0.796875,
    "false_positive": false,
    "start_time": 1662925983.809688,
    "end_time": null,
    "score": 0.71484375,
    "box": [
      200,
      275,
      880,
      731
    ],
    "area": 310080,
    "ratio": 1.4912280701754386,
    "region": [
      54,
      0,
      926,
      872
    ],
    "stationary": true,
    "motionless_count": 51,
    "position_changes": 1,
    "current_zones": [],
    "entered_zones": [
      "driveway"
    ],
    "has_clip": true,
    "has_snapshot": true
  },
  "type": "update"
}
```

**Comment by NickM-27** [comment]
A few things to clear up. 

1. You can check out https://docs.frigate.video/guides/false_positives to see how scores work. Once an object becomes a true positive it will be considered present in a zone even if its score is below the threshold. So since the car isn't a false positive `"false_positive": false` it likely should be showing 2 in your screenshots. 

2. After looking at the data more, I am seeing that the box has a y of 731 which shouldn't be with the change I made (since it seems your camera has a height of 720). Will have to see if I can reproduce it.

**Comment by jmmyerz** [comment]
Let me know what other info might be useful. The car object count for the camera is almost always accurate, but the driveway zone seems to be hit or miss as you can see. Thanks for looking into it!

**Comment by jmmyerz** [comment]
No worries, thank you for looking into it!

**Comment by NickM-27** [comment]
Should be fixed in the just released 0.11 RC3
